src/lavinmq/log_formatter.cr

```diff
@@ -3,13 +3,58 @@ require "log"
 module LavinMQ
   struct JournalLogFormat < ::Log::StaticFormatter
     def run
+      severity
       source
       context(before: '[', after: ']')
       data(before: '[', after: ']')
       string ' '
       message
       exception
     end
+
+    private def severity : Nil
+      @io << '<' << severity_to_priority << '>' << ' '
+    end
+
+    private def exception(*, before = '\n', after = nil) : Nil
+      if ex = @entry.exception
+        @io << '\n'
+        inspect_with_backtrace(ex)
+        @io << after
+      end
+    end
+
+    private def inspect_with_backtrace(ex : Exception) : Nil
+      severity
+      @io << message << " (" << ex.class << ")\n"
+
+      ex.backtrace?.try &.each do |frame|
+        severity
+        @io.print "  from "
+        @io.puts frame
+      end
+
+      if cause = ex.cause
+        severity
+        @io << "Caused by: "
+        inspect_with_backtrace(cause)
+      end
+
+      @io.flush
+    end
+
+    private def severity_to_priority
+      case @entry.severity
+      when ::Log::Severity::Trace  then 7 # journald doesn't have trace
+      when ::Log::Severity::Debug  then 7
+      when ::Log::Severity::Info   then 6
+      when ::Log::Severity::Notice then 5
+      when ::Log::Severity::Warn   then 4
+      when ::Log::Severity::Error  then 3
+      when ::Log::Severity::Fatal  then 2
+      else                              6 # Default to "info"
+      end
```

Comment:
By using `in` instead of `when` you get an "exhaustive case" which means you'll get a compile-time error if you forget to cover any enum member:

```suggestion
      case @entry.severity
      in ::Log::Severity::Trace  then 7 # journald doesn't have trace
      in ::Log::Severity::Debug  then 7
      in ::Log::Severity::Info   then 6
      in ::Log::Severity::Notice then 5
      in ::Log::Severity::Warn   then 4
      in ::Log::Severity::Error  then 3
      in ::Log::Severity::Fatal  then 2
      else                              6 # Default to "info"
      end
```